src/main/scala/com/kinja/config/safeConfig.scala

```diff
@@ -20,169 +20,159 @@ object safeConfig {
     val underlying : Tree = c.prefix.tree match {
       case q"new $_(..$params)" ⇒ params match {
+        case Literal(Constant(i: String)) :: Nil ⇒ Ident(TermName(i))
         case head :: Nil ⇒ head
-        case _           ⇒ c.abort(c.enclosingPosition, "No underlying template given.")
+        case _           ⇒ c.abort(c.enclosingPosition, "No underlying configuration given.")
       }
       case _ ⇒ c.abort(c.enclosingPosition, "Encountered unexpected tree.")
     }
 
-	 val bootupErrors = tq"com.kinja.config.BootupErrors"
-	 val liftedTypesafeConfig = tq"com.kinja.config.LiftedTypesafeConfig"
-
-    // Create a stub for a function in the ConfigApi interface.
-    def apiStub(name : String, typ : Tree) =
-      q"""def ${TermName(name)}(name : String) : $bootupErrors[$typ] = (throw new Exception("")) : $bootupErrors[$typ]"""
-
-	 // The interface of ConfigApi for type-checking.
-	 val configApiStubs = List(
-		q"""val root : $bootupErrors[$liftedTypesafeConfig] = (throw new Exception("")) : $bootupErrors[$liftedTypesafeConfig]""",
-      // format: OFF
-      apiStub("getBoolean",      tq"Boolean"),
-      apiStub("getBooleanList",  tq"List[Boolean]"),
-      apiStub("getConfig",       tq"$liftedTypesafeConfig"),
-      apiStub("getDouble",       tq"Double"),
-      apiStub("getDoubleList",   tq"List[Double]"),
-      apiStub("getDuration",     tq"scala.concurrent.duration.Duration"),
-      apiStub("getDurationList", tq"List[scala.concurrent.duration.Duration]"),
-      apiStub("getInt",          tq"Int"),
-      apiStub("getIntList",      tq"List[Int]"),
-      apiStub("getLong",         tq"Long"),
-      apiStub("getLongList",     tq"List[Long]"),
-      apiStub("getObject",       tq"com.typesafe.config.ConfigObject"),
-      apiStub("getObjectList",   tq"List[com.typesafe.config.ConfigObject]"),
-      apiStub("getString",       tq"String"),
-      apiStub("getStringList",   tq"List[String]"),
-      apiStub("getRawConfig",    tq"com.typesafe.config.Config"))
-      // format: ON
-
-	 // The root element.
-    val root = q"""val root = com.kinja.config.BootupErrors(com.kinja.config.LiftedTypesafeConfig($underlying, "root"))"""
+    val bootupErrors = tq"com.kinja.config.BootupErrors"
+    val liftedTypesafeConfig = tq"com.kinja.config.LiftedTypesafeConfig"
+
+    // The root element.
+    val root = q"""lazy val root = com.kinja.config.BootupErrors(com.kinja.config.LiftedTypesafeConfig($underlying, "root"))"""
+
+    def modBody(impl : Template): Template = {
+      val configApi = tq"com.kinja.config.ConfigApi"
+      val newParents = impl.parents.filter {
+        case Select(Ident(TermName("scala")), TypeName("AnyRef")) ⇒ false
+        case _ ⇒ true
+      } :+ configApi
+
+      // Put members of mixins like ConfigApi into scope.
+      val contextName = freshTerm()
+      val context = q"""val $contextName : Object with ..$newParents = (throw new java.lang.Exception("")) : Object with ..$newParents"""
```

Comment:
Again, this was a bug in the old behavior in that it did not take into account members added by mixins. This solution eliminated the need for `apiStub`s so those were removed.